examples/bind-shopping.amp.html

```diff
@@ -0,0 +1,454 @@
+<!doctype html>
```

Comment:
Cool, this looks like a great start. Some overall notes I have:

> <button bind-attr="{someAttribute: foo}"></button>

We need to make sure this is easy enough to validate. E.g. we can’t allow setting `onclick` this way. We can leave it to runtime, of course. But validator is often handy.

> takes a string with expressions embedded in {} (string interpolation)

I think we should go for double-bracing: `{{}}`, which I think you are doing in most of examples.

> Using custom attributes avoids "flash of unbound content".

Please clarify what this means.

> <amp-img bind-attr="{src: 'http://foo.com/' + bar}"></amp-img>

Even if we go with a expanded form, we should still use string interpolation for all strings. I.e. avoid `+`.

> This approach requires a one-time DOM scan to detect all bindings. This may have performance implications, but may be fine assuming that digest/evaluations are scheduled after initial layout

A. One-time scan is not enough. We will also have to pick up deletes and other mutations via amp-live-list and other components.

B. This scan is potentially very expensive, especially with short-cut forms such as `bind-src`.

That being said, I agree that we should experiment with attribute-based binding. However, I'd like to stress that from the model point of view, these two solutions are equivalent. Thus we first should work on reducing the algebra. I believe we can drastically reduce number of forms and expression syntax.